portfolio/src/main/webapp/script.js

```diff
@@ -473,6 +474,94 @@ function postMarker(lat, lng, name, content) {
   fetch("/markers", options).then(_ => {createMap();});
 }
 
+
+/**
+ * Sends text to servlet to be translated by Cloud Translation API and adds
+ * markers with corresponding translations in infowindows for each country.
+ */
+function translateText() {
+  // Clear previous markers and their InfoWindows.
+  clearCountryMarkers();
+
+  const text = document.getElementById("translate-input").value;
+  
+  if (text.length > 0) {
+    document.getElementById("loading").hidden = false;
+
+    const fetchURL = "translate?text=" + text;
+
+    fetch(fetchURL).then(response => response.json()).then((translationsData) => {
+      document.getElementById("loading").hidden = true;
+      for (translationData of translationsData) {
+        const marker = new google.maps.Marker({
+          position: {lat: translationData.lat, lng: translationData.lng},
+          icon: "http://labs.google.com/ridefinder/images/mm_20_blue.png",
+          map: map
+        }); 
+        
+        const infoWindow = new google.maps.InfoWindow({
+          content: buildTranslationWindowInfo(translationData)
+        });
+
+        marker.addListener("click", () => {
+          infoWindow.open(map, marker);
+        });
+
+        // Store marker so that it can be cleared later
+        countryMarkers.push(marker);
+      }
+    });
+  }
+}
+
+/**
+ * Populates InfoWindow corresponding to country with translation of input 
+ * text in the languages spoken in the country.
+ * @param {Object} translationData Contains country name, languages, and translations
+ * @return {HTML Element} Div containing country name, languages, and translations
+ */
+function buildTranslationWindowInfo(translationData) {
+    const containerDiv = document.createElement("div");
+   
+    const countryElement = document.createElement("b");
+    countryElement.appendChild(document.createTextNode(translationData.country));
+
+    const translationsElement = document.createElement("p");
+    translationsElement.class = "w3-center";
+    
+    for (languageKey of Object.keys(translationData.translations)) {     
+      // LanguageKey is in form 'isoCode englishName'. Below statement extracts englishName
+      const language = languageKey.substr(languageKey.indexOf(" ") + 1);
```

Comment:
I actually did this initially but javscript's split function works differently than java's split function so for a languageKey like "zh-TN Chinese (Traditional)", language becomes "Chinese" instead of "Chinese (Traditional)".

Reply:
I changed the keys in the json file to avoid this problem by making the format "isoCode, englishName" for each language key. These changes have been reflected in the latest commits.